src/pages/workspaces/workspace/WorkspaceContainer.js

```diff
@@ -333,16 +335,16 @@ export const wrapWorkspace = ({ breadcrumbs, activeTab, title, topBarContent, sh
     const workspaceLoaded = !!workspace
 
     const { runtimes, refreshRuntimes, persistentDisks, appDataDisks } = useCloudEnvironmentPolling(googleProject, workspace)
-    const { apps, refreshApps } = useAppPolling(googleProject, name)
+    const { apps, refreshApps } = useAppPolling(googleProject, name, workspace)
     // The following if statements are necessary to support the context bar properly loading runtimes for google/azure
-    // Note that the refreshApps function currently is not supported for azure
     if (workspaceLoaded) {
       if (googleProject !== prevGoogleProject && isGoogleWorkspace(workspace)) {
```

Comment:
I think you can simplify this to the code below. I have no idea though why `refreshRuntimes` is called with true (for "maybeStale" I think) for Azure workspaces, but not for GCP workspaces.

```
if (googleProject !== prevGoogleProject || azureContext !== prevAzureContext) {
    refreshRuntime(isAzureWorkspace(workspace))
    refreshApps()
}

Reply:
@jdcanas perhaps you can explain ^? You added `refreshApps(true)`  back in #2997 and it would be great to add a comment to the code explaining.

Reply:
Yeah I wasn't sure why `refreshRuntimes` was called with true and hence I didn't simplify this. If it is no longer needed I will combine the ifs.

Reply:
@n8landolt @jdcanas can you comment on this?

Reply:
I'm really unsure the reasoning. We'd need Justin's input.

Reply:
@jdcanas can you please provide some feedback?

Reply:
Hmm... I don't know why i would be different. It can't hurt to call it with true for both, it basically forces a refresh on a faster interval as I understand the refresh definition above. 

The simplification you suggest seems fine

Reply:
I have updated the code to reflect Christina's suggestion.

Reply:
It seemed though that @jdcanas was suggesting that it would make sense to call `refreshRuntimes` with true in both cases. It would be nice to clean that up because as it is, it's very mysterious why it is only called with true if the workspace is Azure.